pkg/codegen/docs/gen.go

```diff
@@ -97,30 +121,65 @@ type ConstructorParam struct {
 	DefaultValue string
 }
 
-type resourceArgs struct {
-	Header
+type resourceDocArgs struct {
+	Header header
 
+	// Comment represents the introductory resource comment.
 	Comment  string
 	Examples []exampleUsage
 
 	ConstructorParams map[string]string
 	// ConstructorResource is the resource that is being constructed or
 	// is the result of a constructor-like function.
-	ConstructorResource map[string]PropertyType
-	ArgsRequired        bool
-
-	InputProperties  map[string][]Property
-	OutputProperties map[string][]Property
-	StateInputs      map[string][]Property
-	StateParam       string
+	ConstructorResource map[string]propertyType
+	// ArgsRequired is a flag indicating if the args param is required
+	// when creating a new resource.
+	ArgsRequired bool
+
+	// InputProperties is a map per language and a corresponding slice of
+	// input properties accepted as args while creating a new resource.
+	InputProperties map[string][]property
+	// InputProperties is a map per language and a corresponding slice of
+	// output properties returned when a new instance of the resource is
+	// created.
+	OutputProperties map[string][]property
+
+	// StateInputs is a map per language and the corresponding slice of
+	// state input properties required while looking-up an existing resource.
+	StateInputs map[string][]property
+	// StateParam is the type name of the state param, if any.
+	StateParam string
+
+	// NestedTypes is a slice of the nested types used in the input and
+	// output properties.
+	NestedTypes []docNestedType
+}
 
-	NestedTypes []DocNestedType
+type appearsIn struct {
+	Input  bool
+	Output bool
 }
 
-type stringSet map[string]struct{}
+// stringSet is a type-alias for a map of type tokens
+// and whether or not the type appears in input as well
+// as output properties.
+type stringSet map[string]appearsIn
```

Comment:
Should the type be renamed since it's now more than just a `stringSet`?

Reply:
ha yes!